internal/backend/azure/azure.go

```diff
@@ -33,26 +33,51 @@ const defaultListMaxItems = 5000
 // make sure that *Backend implements backend.Backend
 var _ restic.Backend = &Backend{}
 
-func open(cfg Config, rt http.RoundTripper) (*Backend, error) {
-	debug.Log("open, config %#v", cfg)
+func getCredential(cfg Config) (*storage.SharedKeyCredential, error) {
+	if cfg.AccountKey != "" {
+		cred, err := storage.NewSharedKeyCredential(cfg.AccountName, cfg.AccountKey)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error creating SharedKeyCredential")
+		}
 
-	client, err := storage.NewBasicClient(cfg.AccountName, cfg.AccountKey)
+		return cred, err
+	}
+
+	return nil, errors.New("Not supported")
+}
+
+func getBSU(cfg Config) (storage.ServiceURL, error) {
+	credential, err := getCredential(cfg)
 	if err != nil {
-		return nil, errors.Wrap(err, "NewBasicClient")
+		return storage.ServiceURL{}, err
 	}
 
-	client.HTTPClient = &http.Client{Transport: rt}
+	pipeline := storage.NewPipeline(credential, storage.PipelineOptions{})
+	blobPrimaryURL, err := url.Parse("https://" + credential.AccountName() + ".blob.core.windows.net/")
 
-	service := client.GetBlobService()
+	if err != nil {
+		return storage.ServiceURL{}, err
+	}
+
+	return storage.NewServiceURL(*blobPrimaryURL, pipeline), nil
+}
+
+func open(cfg Config, rt http.RoundTripper) (*Backend, error) {
```

Comment:
The roundTripper is no longer used which breaks debugging and upload/download throttling. Please make sure that http requests use the roundTripper.